File: bedrock_boto/call_bedrock.py
import json
import boto3
import sys

print('Loading function')

def lambda_handler(event, context):
    print("Received event: " + json.dumps(event, indent=2))
    print("Context: " + str(context))
    input_str = event.get('input')
    model_str = event.get('model')

    # Initialize the Bedrock Runtime client
    bedrock_runtime = boto3.client('bedrock-runtime', region_name='us-east-1')
    
    # Define the model ID and input text
    model_id = model_str or 'us.anthropic.claude-3-5-sonnet-20241022-v2:0'
    input_text = input_str or "Please explain the concept of cloud computing."
    
    # Prepare the request body
    request_body = json.dumps({
        "anthropic_version": "bedrock-2023-05-31",
        "max_tokens": 300,
        "messages": [
            {
                "role": "user",
                "content": [
                    {
                        "type": "text",
                        "text": input_text
                    }
                ]
            }
        ],
        "temperature": 0.5,
        "top_p": 1
    })
    
    try:
        # Invoke the Bedrock model
        response = bedrock_runtime.invoke_model(
            modelId=model_id,
            contentType='application/json',
            accept='application/json',
            body=request_body
        )
        
        # Parse and return the response
        response_body = json.loads(response['body'].read())
        
        # Debug print to see the actual response structure
        print("Response body:", json.dumps(response_body, indent=2))
        
        return {
            'statusCode': 200,
            'body': json.dumps(response_body['content'][0]['text'])
        }
    except Exception as e:
        print(f"Error invoking Bedrock model: {str(e)}")
        return {
            'statusCode': 500,
            'body': json.dumps(f"Error: {str(e)}")
        }

def main():
	lambda_handler({}, {})


if __name__ == '__main__':
    sys.exit(main())  # next section explains the use of sys.exit

File: ts_auto/call_bedrock.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.lambdaHandler = lambdaHandler;
const client_bedrock_runtime_1 = require("@aws-sdk/client-bedrock-runtime");
console.log('Loading function');
async function lambdaHandler(event, context) {
    console.log("Received event: " + JSON.stringify(event, null, 2));
    console.log("Context: " + JSON.stringify(context));
    const inputStr = event.input;
    const modelStr = event.model;
    // Initialize the Bedrock Runtime client
    const bedrockRuntime = new client_bedrock_runtime_1.BedrockRuntimeClient({ region: 'us-east-1' });
    // Define the model ID and input text
    const modelId = modelStr || 'us.anthropic.claude-3-5-sonnet-20241022-v2:0';
    const inputText = inputStr || "Please explain the concept of cloud computing.";
    // Prepare the request body
    const requestBody = {
        anthropic_version: "bedrock-2023-05-31",
        max_tokens: 300,
        messages: [
            {
                role: "user",
                content: [
                    {
                        type: "text",
                        text: inputText
                    }
                ]
            }
        ],
        temperature: 0.5,
        top_p: 1
    };
    try {
        // Invoke the Bedrock model
        const command = new client_bedrock_runtime_1.InvokeModelCommand({
            modelId: modelId,
            contentType: 'application/json',
            accept: 'application/json',
            body: JSON.stringify(requestBody)
        });
        const response = await bedrockRuntime.send(command);
        // Parse and return the response
        const responseBody = JSON.parse(new TextDecoder().decode(response.body));
        // Debug print to see the actual response structure
        console.log("Response body:", JSON.stringify(responseBody, null, 2));
        return {
            statusCode: 200,
            body: JSON.stringify(responseBody.content[0].text)
        };
    }
    catch (e) {
        console.error(`Error invoking Bedrock model: ${e}`);
        return {
            statusCode: 500,
            body: JSON.stringify(`Error: ${e}`)
        };
    }
}
async function main() {
    try {
        const result = await lambdaHandler({}, {});
        console.log(result);
    }
    catch (error) {
        console.error(error);
        process.exit(1);
    }
}
if (require.main === module) {
    main().catch(err => {
        console.error(err);
        process.exit(1);
    });
}
